src/modules/explorer/actions.js

```diff
@@ -57,21 +52,33 @@ export const performSearchIfPossible = (datasetID) => (dispatch, getState) => {
 
 export const performIndividualsDownloadCSVIfPossible = (datasetId, individualIds, allSearchResults) =>
     (dispatch, getState) => {
-        console.log("Initiating PerformIndividualsDownloadCSVIfPossible");
-
-        let dataUrl = `${getState().services.itemsByArtifact.metadata.url}/api/individuals?format=csv`;
-
-        // build query string
-        // TODO: This should use the actual JS API for URL construction
+        let ids;
         if (individualIds.length > 0) { // Get only selected results
-            dataUrl += ("&page_size=" + individualIds.length);
-            individualIds.forEach(id => dataUrl += `&id=${id}`);
+            ids = individualIds;
         } else { // Get all search results
-            dataUrl += ("&page_size=" + allSearchResults.length);
-            allSearchResults.forEach(sr => dataUrl += `&id=${sr.key}`);
+            ids = allSearchResults.map(sr => sr.key);
         }
```

Comment:
Nit. and personal preference: we could use a ternary operator here to declare and assign value to `ids` (as a const)

Reply:
U